zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -1007,20 +1017,29 @@ public void updatePersonalizedMode(String noteId,
   public void moveNoteToTrash(String noteId,
                               ServiceContext context,
                               ServiceCallback<Note> callback) throws IOException {
-    Note note = notebook.getNote(noteId);
-    if (note == null) {
-      callback.onFailure(new NoteNotFoundException(noteId), context);
+    if (!checkPermission(noteId, Permission.OWNER, Message.OP.MOVE_NOTE_TO_TRASH, context, callback)) {
       return;
     }
 
-    if (!checkPermission(noteId, Permission.OWNER, Message.OP.MOVE_NOTE_TO_TRASH, context,
-        callback)) {
-      return;
-    }
-    String destNotePath = "/" + NoteManager.TRASH_FOLDER + note.getPath();
+    String destNotePath = "/" + NoteManager.TRASH_FOLDER + notebook.getNoteManager().getNotesInfo().get(noteId);
```

Comment:
Why changing `note.getPath()`to `notebook.getNoteManager().getNotesInfo().get(noteId)`